### PR TITLE
Frozen autoscaling decider based on storage pct

### PR DESCRIPTION
**Pull request by henningandersen**
The frozen tier partially downloads shards only. This commit
introduces an autoscaling decider that scales the total storage
on the tier according to a configurable percentage relative to
the total data set size.

### CODE REVIEW

File: x-pack/plugin/autoscaling/src/main/java/org/elasticsearch/xpack/autoscaling/storage/FrozenStorageDeciderService.java [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+            .mapToLong(imd -> estimateSize(imd, context.info()))
+            .sum();
+
+        long storageSize = (long) (PERCENTAGE.get(configuration) * dataSetSize) / 100;
+        return new AutoscalingDeciderResult(AutoscalingCapacity.builder().total(storageSize, null).build(), new FrozenReason(dataSetSize));
+    }
+
+    static long estimateSize(IndexMetadata imd, ClusterInfo info) {
+        int copies = imd.getNumberOfReplicas() + 1;
+        return IntStream.range(0, imd.getNumberOfShards())
```
Comment (jasontedor) [inline x-pack/plugin/autoscaling/src/main/java/org/elasticsearch/xpack/autoscaling/storage/FrozenStorageDeciderService.java:57]: 🤨

**Review by jasontedor — APPROVED** [review]
LGTM.

File: x-pack/plugin/autoscaling/src/main/java/org/elasticsearch/xpack/autoscaling/storage/FrozenStorageDeciderService.java (hunk 1 above)
Comment (henningandersen) [reply x-pack/plugin/autoscaling/src/main/java/org/elasticsearch/xpack/autoscaling/storage/FrozenStorageDeciderService.java:57]: Are you requesting a change here? Will merge this as is but can change this if you wish in a follow-up.

Comment (henningandersen) [reply x-pack/plugin/autoscaling/src/main/java/org/elasticsearch/xpack/autoscaling/storage/FrozenStorageDeciderService.java]: I changed it into a loop, guessing a bit that this was what you were after.

**Pull request merged**